wgsl/index.bs

```diff
@@ -8627,12 +8632,126 @@ where compatibility is defined by the following table.
 See the [$validating GPUProgrammableStage|WebGPU API$]
 specification for interface validation requirements.
 
+### Buffer Binding Determines Runtime-Sized Array Element Count ### {#buffer-binding-determines-runtime-sized-array-element-count}
+
+When a [=storage buffer=] variable contains a [=runtime-sized=] array, then the number of elements in that array
+is determined from the size of the corresponding {{GPUBufferBinding}}:
+
+<blockquote algorithm="element count of runtime-sized array">
+* Let |T| be the [=store type=] for a [=storage buffer=] variable,
+    where |T| is a [=runtime-sized=] array type or contains a runtime-sized array type.
+* Let |EBS| be the
+    [=effective buffer binding size=]
+    for the {{GPUBufferBinding}} bound to the pipeline binding address corresponding to the storage buffer variable.
+* Then <dfn noexport>NRuntime</sub></dfn>, i.e.
+    the number of elements in the runtime-sized array,
+    is the largest integer such that [=SizeOf=](|T|) &le; |EBS|.
+
+</blockquote>
+
+In general, the [=NRuntime=] for a runtime-size array is:
+<blockquote>
+[=truncate=]((EBBS &minus; *array_offset*) &divide; *array_element_size*), where:
+* EBBS is the [=effective buffer binding size=] associated with the variable,
+* *array_offset* is the byte offset of the runtime-sized array within the store type of the variable.
+    * It is zero if the store type is the runtime-sized array itself.
+    * Otherwise the store type is a structure, and its last member is the runtime-sized array.
+        In this case *array_offset* is the [=OffsetOfMember|byte offset of that member=] within the structure.
+* *array_element_size* is the [=SizeOf|size of=] the element type of the runtime-sized array.
```

Comment:
The array element stride may be larger than the `SizeOf` the element type (e.g. for `vec3`).

Reply:
of course. Thanks!

Reply:
Fixed. PTAL.